platform/android/export/export.cpp

```diff
@@ -2418,13 +2475,40 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 				store_in_gradle_project(img_path, data, Z_NO_COMPRESSION);
 			}
 		}
+	}
+
+	Error gradle_build(const Ref<EditorExportPreset> &p_preset, bool p_debug, const String &p_path, String &src_apk, int p_flags = 0) {
+
+		{ //test that installed build version is alright
+			FileAccessRef f = FileAccess::open("res://android/.build_version", FileAccess::READ);
+			if (!f) {
+				EditorNode::get_singleton()->show_warning(TTR("Trying to build from a custom built template, but no version info for it exists. Please reinstall from the 'Project' menu."));
+				return ERR_UNCONFIGURED;
+			}
+			String version = f->get_line().strip_edges();
+			if (version != VERSION_FULL_CONFIG) {
+				EditorNode::get_singleton()->show_warning(vformat(TTR("Android build version mismatch:\n   Template installed: %s\n   Godot Version: %s\nPlease reinstall Android build template from 'Project' menu."), version, VERSION_FULL_CONFIG));
+				return ERR_UNCONFIGURED;
+			}
+		}
+		//build project if custom build is enabled
+		String sdk_path = EDITOR_GET("export/android/custom_build_sdk_path");
+		ERR_FAIL_COND_V_MSG(sdk_path == "", ERR_UNCONFIGURED, "Android SDK path must be configured in Editor Settings at 'export/android/custom_build_sdk_path'.");
 
-        //TODO: replicate the functionality of _fix_manifest
+        String output_gradle_path = p_path.replace(ProjectSettings::get_singleton()->get_resource_path(), "")
+                                            .replace(".apk","").insert(0, "../../../../../..") + "_${variant.name}.apk";
```

Comment:
"../../../../../.." seems a bit brittle - is there another way to get the path you need?

Reply:
brittle? And I don't think so. I tried using absolute paths but I got an error saying something along the lines of "build.gradle cannot use absolute paths"

Reply:
Can you provide an example (in a code review comment) about what an example of `ProjectSettings::get_singleton()->get_resource_path()` looks like before and after your change?

Reply:
get_resource_path returns the absolute path to the directory where the project.godot file is. p_path is the absolute path of the exported apk.

Reply:
To make this clearer I'd suggest breaking this into a couple lines with more descriptive variable names:

```
String apk_relative_path = p_path.replace(ProjectSettings::get_singleton()->get_resource_path(), "");
String output_gradle_path = <name_here>.replace(".apk","").insert(0, "../../../../../..") + "_${variant.name}.apk";
```
Not sure if `apk_relative_path` is correct. Is `output_gradle_path` the right name for an apk path?